src/sqlfluff/core/rules/base.py

```diff
@@ -1067,11 +1078,42 @@ def eval(self, **kwargs):
                     code, self.name
                 )
             )
-        self._register[code] = dict(code=code, description=description, cls=cls)
+
+        try:
+            groups = cls.groups
+        except AttributeError:
+
+            raise AttributeError(
+                (
+                    "Rule {!r} doesn't belong to any rule groups. "
```

Comment:
This error message implies that every rule must belong to at least one group, but it appears to allow `groups` to be an empty collection. Which of these is our true intention?

Reply:
I think the former. What do you think about explicitly checking that the rule is a part of the 'all' group, and making the error message pertain to that?

Reply:
Sounds good!

Reply:
WDYT about not needing to explicitly specify `all`? Should it be implicit? That way the `groups` config is more of an "additional groups".

Can see positives and negatives:

positives:
- a rule can't be forgotten to be added to `all`
- less needless boilerplate

negatives:
- test would catch the case where it's forgotten
- potentially clearer to list all the groups - including `all`
- documentation might be easier

I think I've veering on keeping it as you have, but definitely think we need a test to check someone doesn't add a new rule and forget to add to `all`.

Reply:
We could update the existing check in this PR, the one that catches `AttributeError`, and also verify the value is not empty.

Reply:
Think it needs to be more than just empty. Need to ensure `all` is in there. Otherwise that could be missed by a rule author.

Reply:
True.

Reply:
I'll keep the attribute error check in case someone forgets to add the groups variable at all, and also add a check to make sure the 'all' group is present